File: classification.py
# classification.py

def classify_by_segment(segment: str) -> str:
    """
    segment の文字列をもとにカテゴリを返す
    完全一致ではなく「含む」で判定する
    """

    if not segment:
        return "other"

    s = segment.strip()

    # ドラッグストアaaa
    if "ドラッグ" in s:
        return "drugstore"

    # コンビニ
    if "コンビニ" in s:
        return "convenience"

    # 喫茶・カフェ・バー・ラウンジ
    if any(k in s for k in ["喫茶", "カフェ", "バー", "ラウンジ"]):
        return "cafe_bar"

    # スーパー・小売
    if any(k in s for k in ["スーパー", "小売", "百貨店"]):
        return "super_retail"

    return "other"

File: static/js/classification.js
// データの業種に対して分類わけ
window.classifyBySegment = function(segment){
  if (!segment) return "restaurant";
  const s = segment.trim();

  if (s.includes("ドラッグ")) return "drugstore";
  if (s.includes("コンビニ")) return "convenience";
  if (s.includes("喫茶") || s.includes("カフェ") || s.includes("バー") || s.includes("ラウンジ"))
    return "cafe";
  if (s.includes("スーパー") || s.includes("小売") || s.includes("百貨店"))
    return "super";

  return "restaurant";
}
